exclause/with.go

```diff
@@ -19,16 +31,56 @@ import (
 //
 //	// WITH RECURSIVE `cte` AS (SELECT * FROM `users`) SELECT * FROM `cte`
 //	db.Clauses(exclause.With{Recursive: true, CTEs: []exclause.CTE{{Name: "cte", Subquery: exclause.Subquery{DB: db.Table("users")}}}}).Table("cte").Scan(&users)
+//
+//	// WITH `cte` AS MATERIALIZED (SELECT * FROM `users`) SELECT * FROM `cte`
+//	db.Clauses(exclause.With{CTEs: []exclause.CTE{{Name: "cte", Subquery: exclause.Subquery{DB: db.Table("users")}, Materialized: exclause.CTEMaterialize}}}).Table("cte").Scan(&users)
+//
+//	// WITH `cte` AS NOT MATERIALIZED (SELECT * FROM `users`) SELECT * FROM `cte`
+//	db.Clauses(exclause.With{CTEs: []exclause.CTE{{Name: "cte", Subquery: exclause.Subquery{DB: db.Table("users")}, Materialized: exclause.CTENotMaterialize}}}).Table("cte").Scan(&users)
+//
+//	// WITH `cte1` AS MATERIALIZED (...), `cte2` AS NOT MATERIALIZED (...) SELECT * FROM `cte1`
+//	db.Clauses(exclause.With{CTEs: []exclause.CTE{
+//		exclause.NewMaterializedCTE("cte1", exclause.Subquery{DB: db.Table("users")}),
+//		exclause.NewNotMaterializedCTE("cte2", exclause.Subquery{DB: db.Table("products")}),
+//	}}).Table("cte1").Scan(&users)
 type With struct {
 	Recursive bool
 	CTEs      []CTE
 }
 
 // CTE common table expressions
 type CTE struct {
-	Name     string
-	Columns  []string
-	Subquery clause.Expression
+	Name         string
+	Columns      []string
+	Subquery     clause.Expression
+	Materialized CTEMaterializeOption
+}
+
+// NewCTE creates a new CTE with unspecified materialization (database default)
+func NewCTE(name string, subquery clause.Expression) CTE {
```

Comment:
I'd like to match usage with other  `NewXxxx` methods. This is because I don't think most GORM users understand the  `clause.Expression` type.
Is there an important intention behind using  `clause.Expression`? If not, I want the following arguments!

```go
func NewCTE(name string, subquery interface{}, args ...interface{}) CTE {
    ...
}
```